crates/chain/src/chain_data.rs

```diff
@@ -15,25 +15,45 @@ use crate::Anchor;
     ))
 )]
 pub enum ChainPosition<A> {
-    /// The chain data is seen as confirmed, and in anchored by `A`.
-    Confirmed(A),
-    /// The chain data is not confirmed and last seen in the mempool at this timestamp.
-    Unconfirmed(u64),
+    /// The chain data is confirmed as it is anchored in the best chain by `A`.
+    Confirmed {
+        /// The [`Anchor`].
+        anchor: A,
+        /// Whether the chain data is anchored transitively by a child transaction.
+        ///
+        /// If the value is `Some`, it means we have incomplete data. We can only deduce that the
+        /// chain data is confirmed at a block equal to or lower than the block referenced by `A`.
+        transitively: Option<Txid>,
```

Comment:
⛏️ I think calling the field either "child" or "descendant" would be best.

Reply:
I think `transitively` is still more descriptive. If it is `Some`, we know that the tx is "confirmed transitively".

If we name it "child" or "descendant", my first thought when looking at it is; child of what? How does this relate to the anchor?

Reply:
I'm on the fence, @LagginTimes what do you think?

Reply:
I think in this naming scheme, `anchor` / `transitively` makes more sense to me. 🤔